[Cosmos] Connection Retry Policy samples (#23945)
Azure/azure-sdk-for-python: sdk/cosmos/azure-cosmos/CHANGELOG.md

#### Bugs Fixed

#### Other Changes

>  - Added sample for retry config for the sync and async clients
>  - Deprecated the options connection_retry_policy and retry_options in the sync client

Reviewer: ```suggestion
 - Added samples showing how to configure retry options for both the sync and async clients.
 - Deprecated the `connection_retry_policy` and `retry_options` options in the sync client.
```
